src/disjoint_mut.rs

```diff
@@ -54,6 +59,21 @@ pub struct DisjointMutGuard<'a, T: AsMutPtr, V: ?Sized> {
     range: (Bound<usize>, Bound<usize>),
 }
 
+impl<'a, T: AsMutPtr> DisjointMutGuard<'a, T, [u8]> {
+    pub fn cast<V: AsBytes + FromBytes>(self) -> DisjointMutGuard<'a, T, V> {
+        let mut old_guard = ManuallyDrop::new(self);
```

Comment:
Is this because you can't move out of a type that `impl Drop`s?

Reply:
Yes, exactly. We need to not drop the guard, as we are replacing it.

Reply:
Could you add that as a comment?  Otherwise it's all good.